src/shell_minimal.js

```diff
@@ -26,6 +39,17 @@ if (ENVIRONMENT_IS_NODE) {
 }
 #endif
 
+#if ENVIRONMENT_MAY_BE_SHELL
+if (ENVIRONMENT_IS_SHELL) {
+#if WASM
+  Module['wasm'] = read('{{{ TARGET_BASENAME }}}.wasm', 'binary');
+#else
+  eval(read('{{{ TARGET_BASENAME }}}.asm.js')+'');
+  Module['mem'] = read('{{{ TARGET_BASENAME }}}.mem', 'binary');
```

Comment:
Do v8/SpiderMonkey not have `__dirname` type of construct? Or is `read()` always relative to the current script that's executed (instead of being relative to current working directory?). If not, then this would turn into `read('a.asm.js')` so if one is running a script from another directory than cwd, this lookup would fail?

Reply:
Yeah. These shells are mostly useful for testing and benchmarking. We can't do stuff like __dirname in them AFAIK.